components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java

```diff
@@ -371,8 +377,36 @@ public void deleteRole(String roleName) throws UserAdminException {
      * @throws UserAdminException
      */
     public void updateRoleName(String roleName, String newRoleName) throws UserAdminException {
+
         try {
-            getUserAdminProxy().updateRoleName(roleName, newRoleName);
+            boolean isInternalRole = false;
+            if (roleName.startsWith(UserCoreConstants.INTERNAL_DOMAIN + UserCoreConstants.DOMAIN_SEPARATOR) ||
+                    roleName.startsWith(UserCoreConstants.APPLICATION_DOMAIN + UserCoreConstants.DOMAIN_SEPARATOR)) {
+                isInternalRole = true;
+            }
```

Comment:
Can't we use the existing [isInternalRole(String roleName)](https://github.com/wso2/carbon-identity-framework/blob/master/components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java#L536-L545) method for this?